plopfile.js

```diff
@@ -0,0 +1,320 @@
+    description: 'Generate a Component either using ES6 class or Stateless Function',
+    prompts: [{
+      type: 'list',
+      name: 'type',
+      message: 'Select the type of component',
+      choices: () => ['ES6 Class', 'Stateless Function']
+    }, {
+      type: 'input',
+      name: 'name',
+      message: 'What should it be called?',
+      validate: value => {
+        if ((/.+/).test(value)) {
+          return true;
+        }
+        return 'name is required';
+      }
+    }, {
+      type: 'input',
+      name: 'description',
+      message: 'Describe what the component does?',
+      validate: value => {
+        if ((/.+/).test(value)) {
+          return true;
+        }
+        return 'description is required';
+      }
+    }, {
+      type: 'confirm',
+      name: 'wantCSS',
+      message: 'Do you want to create corresponding CSS file?'
+    }],
+    actions: data => {
+      const actions = [{
+        type: 'add',
+        path: 'app/components/{{properCase name}}/index.js',
+        templateFile: data.type === 'ES6 Class' ? './webpack/templates/component/es6.js.hbs' : './webpack/templates/component/stateless.js.hbs',
+        abortOnFail: true
+      }, {
+        type: 'add',
+        path: 'app/components/{{properCase name}}/{{properCase name}}.test.js',
+        templateFile: './webpack/templates/component/component.test.js.hbs',
+        abortOnFail: true
+      }];
+
+      if (data.wantCSS) {
+        actions.push({
+          type: 'add',
+          path: 'app/components/{{properCase name}}/styles.css',
+          templateFile: './webpack/templates/component/styles.css.hbs',
+          abortOnFail: true
+        });
+      }
+
+      return actions;
+    }
+  });
+
+  // Container Component Generator
+  plop.setGenerator('container', {
+    description: 'Generate a Container Component',
+    prompts: [{
+      type: 'input',
+      name: 'name',
+      message: 'What should it be called?',
+      validate: value => {
+        if ((/.+/).test(value)) {
+          return true;
+        }
+        return 'name is required';
+      }
+    }, {
+      type: 'input',
+      name: 'description',
+      message: 'Describe what the container component does?',
+      validate: value => {
+        if ((/.+/).test(value)) {
+          return true;
+        }
+        return 'description is required';
+      }
+    }, {
+      type: 'confirm',
+      name: 'wantCSS',
+      message: 'Do you want to create corresponding CSS file?'
+    }, {
+      type: 'confirm',
+      name: 'connectRedux',
+      message: 'Do you want to connect the page to a redux store/dispatch actions?'
+    }, {
+      type: 'confirm',
+      name: 'wantActionsAndReducer',
+      message: 'Do you want to generate corresponding actions/reducer files for the container?',
+      when: answers => answers.connectRedux
+    }, {
+      type: 'confirm',
+      name: 'wantSelector',
+      message: 'Do you want to add selectors to the connect method?',
```

Comment:
Did you choose `Select from already available selectors` or `Generate new one`?

If you chose the first option, you can select multiple selectors from the list which you want to add to the `createSelector` method.

If you chose the second one, you have to enter name and description, it will be added to `createSelector` method.

I just checked both of them. They are working as expected.